packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -730,6 +732,20 @@ export class BasicCrawler<
                     requestQueue: await this.getRequestQueue(),
                 });
             },
+            sendRequest: async (options?: OptionsInit) => {
+                return gotScraping({
+                    url: request!.url,
+                    method: request!.method,
+                    body: request!.payload,
+                    headers: request!.headers,
+                    ...options,
```

Comment:
What about cookies? There are two possible options.

1. Pass a cookieJar

```ts
async requestHandler({ sendRequest, cookieJar }) {
    const response = await sendRequest({ cookieJar }) as Response<string>;
	...
},
```

2. Pass a cookie header (fire-and-forget)

```ts
async requestHandler({ sendRequest, cookieJar, request }) {
    const response = await sendRequest({
		headers: {
				cookie: cookieJar.getCookieStringSync(request.url),
		},
	}) as Response<string>;
	...
},
```

I'm against the latter because it doesn't work when redirecting back and forth (so this may transform into an anti-pattern). Also currently we cannot access `session.cookieJar` because it's private.